src/lib/containers/download_list_v2/DownloadListActionsRequired.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useEffect } from 'react'
+import { useErrorHandler } from 'react-error-boundary'
+import { toError } from '../../utils/ErrorUtils'
+import { useGetDownloadListActionsRequiredInfinite } from '../../utils/hooks/SynapseAPI/useGetDownloadListActionsRequired'
+import { useInView } from 'react-intersection-observer'
+import { ActionRequiredResponse } from '../../utils/synapseTypes/DownloadListV2/QueryResponseDetails'
+import { SynapseSpinner } from '../LoadingScreen'
+import { ActionRequiredCount, MeetAccessRequirement, RequestDownload } from '../../utils/synapseTypes/DownloadListV2/ActionRequiredCount'
+import { MeetAccessRequirementCard } from './MeetAccessRequirementCard'
+import { RequestDownloadCard } from './RequestDownloadCard'
+
+export default function DownloadListActionsRequired() {
+  const handleError = useErrorHandler()
+  // Load the next page when this ref comes into view.
+  const { ref, inView } = useInView()
+  const {
+    data,
+    status,
+    isFetching,
+    hasNextPage,
+    fetchNextPage,
+    isError,
+    error: newError,
+    refetch,
+  } = useGetDownloadListActionsRequiredInfinite()
+
+  useEffect(() => {
+    if (isError && newError) {
+      handleError(toError(newError))
+    }
+  }, [isError, newError, handleError])
+
+  useEffect(() => {
+    if (
+      status === 'success' &&
+      !isFetching &&
+      hasNextPage &&
+      fetchNextPage &&
+      inView
+    ) {
+      fetchNextPage()
+    }
+  }, [status, isFetching, hasNextPage, fetchNextPage, inView])
+
+  const allRows = data
+    ? ([] as ActionRequiredCount[]).concat.apply(
+        [],
+        data.pages.map(
+          page => (page.responseDetails as ActionRequiredResponse).page,
+        ),
+      )
+    : []
```

Comment:
I think I came up with this monstrosity, maybe I should come up with something that makes more sense 🙂